kratos/includes/properties.h

```diff
@@ -277,30 +295,138 @@ class Properties : public IndexedObject
+     * @return The current number of subproperties
+     */
+    std::size_t NumberOfSubproperties()
+    {
+        return mSubPropertiesList.size();
+    }
+
+    /**
+     * @brief This method insert a new property into the list of subproperties
+     * @param pNewSubProperty The new property to be added
+     */
+    void AddSubProperty(Properties::Pointer pNewSubProperty)
+    {
+        mSubPropertiesList.insert(mSubPropertiesList.begin(), pNewSubProperty);
+    }
+
+    /**
+     * @brief This method gets the subproperty from the index corresponding to the property id
+     * @param SubPropertyIndex The index of the subproperty to be get
+     * @return The pointer to the subproperty of interest
+     */
+    Properties::Pointer pGetSubProperty(IndexType SubPropertyIndex)
+    {
+        return mSubPropertiesList(SubPropertyIndex);
+    }
+
+    /**
+     * @brief This method gets the subproperty from the index corresponding to the property id
+     * @param SubPropertyIndex The index of the subproperty to be get
+     * @return The reference to the subproperty of interest
+     */
+    Properties& GetSubProperty(IndexType SubPropertyIndex)
+    {
+        return *(mSubPropertiesList(SubPropertyIndex));
+    }
+
+    /**
+     * @brief This method gets the subproperty from the index corresponding to the property id
+     * @param SubPropertyIndex The index of the subproperty to be get
+     * @return The reference to the subproperty of interest
+     */
+    Properties& GetSubProperty(IndexType SubPropertyIndex) const
+    {
+        return *(mSubPropertiesList.find(SubPropertyIndex));
+    }
+
+    /**
+     * @brief This method returns the whole list of subproperties
+     * @return The whole lis of subproperties
+     */
+    PropertiesContainerType& GetSubProperties()
+    {
+        return SubPropetiesList();
+    }
+
+    /**
+     * @brief This method returns the whole list of subproperties
+     * @return The whole lis of subproperties
+     */
+    PropertiesContainerType const& GetSubProperties() const
+    {
+        return SubPropetiesList();
+    }
+
+    /**
+     * @brief This method set the whole list of subproperties
+     * @param rSubPropetiesList The list of subproperties
+     */
+    void SetSubProperties(PropertiesContainerType& rSubPropetiesList)
+    {
+        mSubPropertiesList = rSubPropetiesList;
+    }
+
     ///@}
     ///@name Access
     ///@{
 
+    /**
+     * @brief This method returns the whole data container
+     * @return Data container
+     */
     ContainerType& Data()
     {
         return mData;
     }
 
+    /**
+     * @brief This method returns the whole data container (constant)
+     * @return Data container
+     */
     ContainerType const& Data() const
     {
         return mData;
     }
 
+    /**
+     * @brief This method returns the whole list of subproperties (constant)
+     * @return The whole lis of subproperties
+     */
+    PropertiesContainerType& SubPropetiesList()
```

Comment:
I think for consistency with the rest of Kratos we shall call this GetSub...

it is true on the other side that for example Tables does not comply with this naming. @pooyan-dadvand opinions?

Reply:
This method already exists:

https://github.com/KratosMultiphysics/Kratos/blob/80332ee8dec355c6d78e9d153df521c68f90d000/kratos/includes/properties.h#L350

Reply:
BTW there is a typo SubPropeties (lacks an r)

Reply:
This is an access method, consistent with Data() (not called GetData)

Reply:
@pooyan-dadvand this is on you.

Reply:
I would keep this method private until we have a clear use case for it....